addons/website/tools.py

```diff
@@ -169,6 +169,16 @@ def text_from_html(html_fragment, collapse_whitespace=False):
     """
     # lxml requires one single root element
     tree = etree.fromstring('<p>%s</p>' % html_fragment, etree.XMLParser(recover=True))
+
+    # Remove scripts or other technical elements that should not be converted
+    # into text.
+    xpath_filters = [
+        '//script',
+        '//*[@class="css_non_editable_mode_hidden"]'
+    ]
```

Comment:
@bvr-odoo That's fine with me. However, for added security, wouldn't it be safer to include '//style' as well? In case of custom css injected?

Reply:
> @bvr-odoo That's fine with me. However, for added security, wouldn't it be safer to include '//style' as well? In case of custom css injected?

Maybe, good point